addons/web_editor/static/lib/odoo-editor/src/commands/deleteForward.js

```diff
@@ -14,21 +13,78 @@ import {
     rightLeafOnlyNotBlockNotEditablePath,
     rightLeafOnlyPathNotBlockNotEditablePath,
     isNotEditableNode,
-} from '../utils/utils.js';
+    splitTextNode,
+    prepareUpdate,
+    isVisibleStr,
+    isInPre,
+    fillEmpty,
+    setSelection,
+    isZWS,
+    childNodeIndex, boundariesOut
+} from "../utils/utils.js";
 
-Text.prototype.oDeleteForward = function (offset) {
-    if (offset < this.length) {
-        this.oDeleteBackward(offset + 1);
-    } else {
-        HTMLElement.prototype.oDeleteForward.call(this.parentNode, childNodeIndex(this) + 1);
+Text.prototype.oDeleteForward = function (offset, alreadyMoved = false) {
+    const parentNode = this.parentNode;
+
+    if (offset === this.nodeValue.length) {
+        // Delete at the end of a text node is not a specific case to
+        // handle, let the element implementation handle it.
+        HTMLElement.prototype.oDeleteForward.call(this, offset, alreadyMoved);
+        return;
     }
+
+    // Get the size of the unicode character to remove.
+    const charSize = [...this.nodeValue.slice(0, offset + 1)].pop().length;
```

Comment:
Honestly I don't know. I just "copied" this code sniped from `deleteBackward.js > ln:38`.

I haven't tested it. 
But it should work as well as before, since I've reproduced the logic that was previously applied in `Text.prototype.oDeleteBackward` into `Text.prototype.oDeleteForward`.